### PR TITLE
chore: convert assertRequiredOptions to TS

### PR DESCRIPTION
**Pull request by konojunya**
Summary:
---------

Part of #683 
Converted `commands/tools/assertRequiredOptions.js` to typescript

Test Plan:
----------

`yarn test`

### CODE REVIEW

File: packages/cli/src/tools/assertRequiredOptions.ts [hunk 1]
```diff
@@ -6,14 +6,16 @@
...
-
 import {Option} from 'commander';
 import {camelCase} from 'lodash';
 
 // Commander.js has a 2 years old open issue to support <...> syntax
 // for options. Until that gets merged, we run the checks manually
 // https://github.com/tj/commander.js/issues/230
-export default function assertRequiredOptions(options, passedOptions) {
+export default function assertRequiredOptions(
+  options: any[],
```
Comment (thymikee) [inline packages/cli/src/tools/assertRequiredOptions.ts]: use `Command['options']`. You'll find `Command` in `@react-native-community/cli-types`

File: packages/cli/src/tools/assertRequiredOptions.ts [hunk 2]
```diff
@@ -6,14 +6,16 @@
...
+  passedOptions: any,
```
Comment (thymikee) [inline packages/cli/src/tools/assertRequiredOptions.ts]: let's make this a `Record<string, unknown>` or something similar, but not `any` :)

**Comment by konojunya** [comment]
@thymikee fixup :)

File: packages/cli/src/tools/assertRequiredOptions.ts [hunk 3]
```diff
@@ -6,14 +6,22 @@
...
  * @format
  */
-
 import {Option} from 'commander';
 import {camelCase} from 'lodash';
+import {Command} from '@react-native-community/cli-types';
+
+type Options = NonNullable<Command['options']>;
+type PassedOptions = {
+  [x: string]: unknown;
```
Comment (thymikee) [inline packages/cli/src/tools/assertRequiredOptions.ts]: Let's be descriptive where we can
```suggestion
  [key: string]: unknown;
```

**Pull request merged**